client/src/main/java/io/pravega/client/tables/impl/KeyVersion.java

```diff
@@ -9,16 +9,45 @@
  */
 package io.pravega.client.tables.impl;
 
+import java.io.Serializable;
 import java.nio.ByteBuffer;
 
 /**
  * Version of a Key in a Table.
  */
-interface KeyVersion {
+public interface KeyVersion extends Serializable {
```

Comment:
I'd vote for our custom serializer. There can be version skews between client and server.

Reply:
I still see java serialization being used hence unresolving.

Reply:
yes, since we have over-ridden the method `io.pravega.client.stream.impl.StreamCutImpl#writeReplace` our custom serialization will be used for KeyVersion.

Reply:
I don't understand @shrids ' comment. I also would like to understand why we are using Java serialization here.

Reply:
We are overriding the `writeReplace` method to ensure our custom serialization is used. ref: https://docs.oracle.com/javase/8/docs/platform/serialization/spec/output.html#a5324

Reply:
Yes, the writeReplace implementation I was referring to is in `KeyVersionImpl`. (updated my previous comment.)